Documentation/ExtensionArchitecture/ExtendingTca/Examples/Index.rst

```diff
@@ -1,84 +1,95 @@
-
-	// Add some fields to FE Users table to show TCA fields definitions
-	// USAGE: TCA Reference > $GLOBALS['TCA'] array reference > ['columns'][fieldname]['config'] / TYPE: "select"
-	$temporaryColumns = array (
-		'tx_examples_options' => array (
-			'exclude' => 0,
-			'label' => 'LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options',
-			'config' => array (
-				'type' => 'select',
-				'items' => array (
-					array('LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options.I.0', '1'),
-					array('LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options.I.1', '2'),
-					array('LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options.I.2', '--div--'),
-					array('LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options.I.3', '3'),
-				),
-				'size' => 1,
-				'maxitems' => 1,
-			)
-		),
-		'tx_examples_special' => array (
-			'exclude' => 0,
-			'label' => 'LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_special',
-			'config' => array (
-				'type' => 'user',
-				'size' => '30',
-				'userFunc' => 'Documentation\\Examples\\Userfuncs\\Tca->specialField',
-				'parameters' => array(
-					'color' => 'blue'
-				)
-			)
-		),
-	);
+   <?php
+   defined('TYPO3') or die();
+
+   // Add some fields to fe_users table to show TCA fields definitions
+   $temporaryColumns = [
+      'tx_examples_options' => [
+         'exclude' => 0,
+         'label' => 'LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options',
+         'config' => [
+            'type' => 'select',
+            'renderType' => 'selectSingle',
+            'items' => [
+               ['',0,],
+               ['LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options.I.0',1,],
+               ['LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options.I.1',2,],
+               ['LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options.I.2','--div--',],
+               ['LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_options.I.3',3,],
+            ],
+            'size' => 1,
+            'maxitems' => 1,
+         ],
+      ],
+      'tx_examples_special' => [
+         'exclude' => 0,
+         'label' => 'LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_special',
+         'config' => [
+            'type' => 'user',
+            // renderType needs to be registered in ext_localconf.php
+            'renderType' => 'specialField',
+            'parameters' => [
+               'size' => '30',
+               'color' => '#F49700',
+            ],
+         ],
+      ],
+   ];
+
+   \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addTCAcolumns('fe_users', $temporaryColumns);
```

Comment:
We should directly put the array as 2nd argument here. I never understood why a temporary variable was used if there is no recycling / customization with another call afterwards.

Reply:
Did so an rephrased the explanation of the example while doing it.